eng/pipelines/templates/jobs/ci.yml

```diff
@@ -189,6 +192,19 @@ jobs:
            -SpellCheckPublicApiSurface
           pwsh: true
           failOnStderr: false
+      - script: >-
+          dotnet build eng/service.proj -warnaserror
+          /t:rebuild
+          /p:DebugType=none
+          /p:SDKType=${{ parameters.SDKType }}
+          /p:ServiceDirectory=${{ parameters.ServiceDirectory }}
+          /p:IncludePerf=false
+          /p:IncludeStress=false
+          /p:PublicSign=false
+          /p:Configuration=$(BuildConfiguration)
+          /p:DefineConstants=SNIPPET
```

Comment:
I think I would pass another property that would add SNIPPET to the define list that already exists instead of overriding DefineConstants for everyone.